master/internal/rm/multirm/multirm.go

```diff
@@ -234,12 +234,33 @@ func (m *MultiRMRouter) GetJobQ(rpName rm.ResourcePoolName) (map[model.JobID]*sp
 func (m *MultiRMRouter) GetJobQueueStatsRequest(req *apiv1.GetJobQueueStatsRequest) (
 	*apiv1.GetJobQueueStatsResponse, error,
 ) {
-	resolvedRMName, err := m.getRM(rm.ResourcePoolName(req.ResourcePools[0]))
+	filteredRMs := map[string]rm.ResourceManager{}
+	for _, rName := range req.ResourcePools {
+		resolvedRMName, err := m.getRM(rm.ResourcePoolName(rName))
+		if err != nil {
+			return nil, err
+		}
+		filteredRMs[rName] = m.rms[resolvedRMName]
+	}
+
+	tmpM := &MultiRMRouter{
+		defaultRMName: m.defaultRMName,
+		syslog:        m.syslog,
+		rms:           filteredRMs,
+	}
+
+	res, err := fanOutRMCall(tmpM, func(rm rm.ResourceManager) (*apiv1.GetJobQueueStatsResponse, error) {
+		return rm.GetJobQueueStatsRequest(req)
+	})
```

Comment:
I feel like making a new router is bug prone

What about something like this instead?
```suggestion
       managersWithRequestedPools := make(map[rm.ResourceManager]bool) // TODO fill in from m.getRM(rm.ResourcePoolName)
	res, err := fanOutRMCall(tmpM, func(rm rm.ResourceManager) (*apiv1.GetJobQueueStatsResponse, error) {
	        if !managersWithRequestedPools[rm] {
	            return rm.GetJobQueueStatsRequest(req) 
	        }
		return &apiv1.GetJobQueueStatsResponse{}
	})
```